File: menu/ui.py
from enum import Enum

import shared
from shared import get_image_path

IMAGES_DIR = shared.get_images_dir(__file__)


class ImagesMapping(str, Enum):
    menu_base = get_image_path(IMAGES_DIR, "Menu_base.bmp")
    menu_refresh = get_image_path(IMAGES_DIR, "Menu_refresh.bmp")
    menu_shutdown = get_image_path(IMAGES_DIR, "Menu_shutdown.bmp")
    menu_restart = get_image_path(IMAGES_DIR, "Menu_restart.bmp")

File: v2/menu/ui.go
package menu

import (
	"embed"
	"github.com/nathan-coates/desk_controller/v2/shared"
	"log"
)

//go:embed *.bmp
var images embed.FS

var (
	MenuBase     string
	MenuRefresh  string
	MenuRestart  string
	MenuShutdown string
)

func init() {
	mapping, err := shared.WriteImagesToFile(images)
	if err != nil {
		log.Fatal(err)
	}

	MenuBase = mapping["Menu_base.bmp"]
	MenuRefresh = mapping["Menu_refresh.bmp"]
	MenuRestart = mapping["Menu_restart.bmp"]
	MenuShutdown = mapping["Menu_shutdown.bmp"]
}
